From 51f0893b2683b4ed48b5b768ca4a55beefacb173 Mon Sep 17 00:00:00 2001
From: "F. von Gellhorn" <flinux@vongellhorn.ch>
Date: Tue, 15 Jun 2021 16:38:10 +0200
Subject: [PATCH 1/1] Repalce Freenode with Libera

Solus has moved to Libera.Chat. Make Libera.Chat the default network
---
 data/org.kde.konversation.appdata.xml | 62 +++++++++++++--------------
 src/config/preferences.cpp            |  4 +-
 src/irc/servergroupdialogui.ui        |  2 +-
 src/irc/serverlistdialogui.ui         |  2 +-
 4 files changed, 35 insertions(+), 35 deletions(-)

diff --git a/data/org.kde.konversation.appdata.xml b/data/org.kde.konversation.appdata.xml
index 8ef253e..5a9383b 100644
--- a/data/org.kde.konversation.appdata.xml
+++ b/data/org.kde.konversation.appdata.xml
@@ -78,43 +78,43 @@
   <summary xml:lang="zh-TW">IRC 用戶端程式</summary>
   <description>
     <p>
-      Konversation is an IRC client which gives you speedy access to Freenode network's channels,
+      Konversation is an IRC client which gives you speedy access to Libera network's channels,
       where you will find support for most distributions. Its defaults are such that it needs no
       configuration other than telling it the name of your distribution channel, to get started.
     </p>
     <p xml:lang="ar">«محادثك» هو عميل آي‌آرسي يعطيك وصول سريع إلى قنوات شبكة «فرينود»، حيث ستجد الدّعم لمختلف التّوزيعات. الافتراضيّات فيه مضبوطة بحيث لا تحتاج لأن تفعل شيئا عدا كتابة اسم قناة التّوزيعة، ومن ثمّ تبدأ الدّردشة.</p>
-    <p xml:lang="ca">El Konversation és un client d'IRC que dóna accés ràpid als canals de la xarxa Freenode, a on trobareu ajuda per a la majoria de les distribucions. De manera predeterminada, per a començar no necessita més configuració que indicar el nom del canal de la distribució.</p>
-    <p xml:lang="ca-valencia">El Konversation és un client d'IRC que dóna accés ràpid als canals de la xarxa Freenode, a on trobareu ajuda per a la majoria de les distribucions. De manera predeterminada, per a començar no necessita més configuració que indicar el nom del canal de la distribució.</p>
-    <p xml:lang="da">Konversation er en IRC-klient som giver dig hurtig adgang til kanaler på Freenode-netværket, hvor du finder support til de fleste distributioner. Dets standardindstillinger er sådan at der ikke kræves anden konfiguration end at angive navnet på distributionskanalen for at komme i gang.</p>
-    <p xml:lang="de">Konversation ist ein IRC-Programm, mit dem Sie schnell auf Freenode-Kanäle zugreifen können, in denen Sie Unterstützung für die meisten Distributionen erhalten. Die Voreinstellungen sind so gewählt, dass Sie nur den Kanalnamen Ihrer Distribution eingeben müssen.</p>
-    <p xml:lang="el">Το Konversation είναι μια εφαρμογή πελάτη IRC, το οποίο παρέχει ταχεία πρόσβαση σε δικτυακά κανάλια Freenode, όπου θα βρείτε υποστήριξη για την πλειοψηφία των διανομών. Οι προκαθορισμένες ρυθμίσεις του καθιστούν περιττή την επιπλέον διαμόρφωση, με εξαίρεση το όνομα του καναλιού της διανομής σας για να ξεκινήσει.</p>
-    <p xml:lang="en-GB">Konversation is an IRC client which gives you speedy access to Freenode network's channels, where you will find support for most distributions. Its defaults are such that it needs no configuration other than telling it the name of your distribution channel, to get started.</p>
-    <p xml:lang="es">Konversation es un cliente de IRC que le proporciona un acceso rápido a los canales de la red Freenode, donde usted encontrará apoyo para muchas distribuciones. Sus opciones por omisión son tales que no necesita otra configuración que decirle el nombre del canal de distribución para empezar.</p>
-    <p xml:lang="et">Konversation on IRC klient, mis lubab sul edukalt tegutseda Freenode'i võrgu kanalitel. Rakenduse vaikimisi seadistused on niisugused, et sul pole alustamiseks üldjuhul vaja midagi rohkemat, kui ainult teada kanalit, millega ühineda.</p>
-    <p xml:lang="fi">Konversation on IRC-asiakas, jolla pääset nopeasti Freenoden verkkokanaville, mistä löydät useimpien jakelujen tuen. Sen oletusasetukset ovat sellaiset, ettei sinun tarvitse kuin vain kertoa jakelusi kanava päästäksesi alkuun.</p>
-    <p xml:lang="fr">Konversation est un client « IRC » vous donnant un accès rapide aux canaux du réseau Freenode où vous trouverez de l'aide pour la plupart des distributions. Ses configurations par défaut sont telles que vous devez uniquement spécifier le nom de votre distribution pour débuter.</p>
-    <p xml:lang="gl">Konversation é un cliente de IRC que lle da acceso rápido ás canles da rede Freenode, onde atopará asistencia técnica para a meirande parte das distribucións. A configuración predeterminada está deseñada para que só teña que indicar o nome da canle da distribución para comezar.</p>
-    <p xml:lang="he">שיחה KDE זה לקוח IRC שנותן גישה מהירה לערוצי רשתות Freenode, ותומך ברוב ההפצות. כברירת מחדל לא צריך להגדיר חוץ לבחור שם של אחד מערוצי ההפצות בשביל להתחיל.</p>
-    <p xml:lang="id">Konversation adalah klien IRC yang memberimu akses cepat ke channel jaringan Freenode, di mana kamu akan menemukan dukungan untuk sebagian besar distribusi. Default-nya adalah tidak diperlukannya konfigurasi selain memberi tahu nama channel distribusimu, untuk memulai.</p>
-    <p xml:lang="it">Konversation è un client IRC che fornisce un acceso rapido ai canali della rete Freenode, dove troverai supporto per la maggior parte delle distribuzioni. I suoi valori predefiniti fanno sì che non ci sia bisogno di configurazione ulteriore a parte il nome del canale della distribuzione per poter iniziare.</p>
-    <p xml:lang="ko">Konversation은 여러 IRC 네트워크 채널에 빠르게 접근할 수 있는 IRC 클라이언트입니다. 대부분 리눅스 배포판 사용자와 대화할 수 있는 Freenode 네트워크 및 배포판 채널을 기본 설정으로 포함합니다.</p>
-    <p xml:lang="nl">Konversation is een IRC-client die u snel toegang geeft tot networkkanalen van Freenode, waar u ondersteuning voor de meeste distributies zult vinden. Zijn standaarden zijn zo dat het geen instellingen nodig heeft anders dan het de naam van uw distributiekanaal te vertellen om van start te gaan.</p>
-    <p xml:lang="nn">Konversation er ein IRC-klient som gjev deg rask tilgang til alle Freenode-kanalane, der du finn støtte for dei fleste distribusjonane. Standardverdiane er slik at det ikkje trengst noko vidare oppsett; du treng berre oppgje kanalen til distribusjonen din for å komma i gang.</p>
-    <p xml:lang="pl">Konversation jest klientem IRC, który szybko daje ci dostęp do kanałów sieci Freenode, gdzie mozesz znaleźć wsparcie dla większości dystrybucji. Dzięki jego ustawieniom domyślnym wystarczy, że podasz nazwę swojego kanału dystrybucji, aby zacząć go używać.</p>
-    <p xml:lang="pt">O Konversation é um cliente de IRC que lhe dá um acesso rápido aos canais da rede Freenode, onde poderá obter suporte para a maioria das distribuições. Tem um conjunto de definições por omissão suficientemente úteis que praticamente não necessita de mais configurações do que definir o nome do canal da sua distribuição para começar.</p>
-    <p xml:lang="pt-BR">O Konversation é um cliente IRC que fornece a você acesso rápido aos canais da rede Freenode, onde você irá encontrar suporte para a maioria das distribuições. Seus padrões são tais que ele não precisa de nenhuma configuração além de dizer o nome do canal de sua distribuição para poder começar.</p>
-    <p xml:lang="sk">Konversation je IRC klient, ktorý vám umožní rýchly prístup ku kanálom siete Freenode, kde nájdete podporu pre väčšinu distribúcií. Predvolene nepotrebuje žiadne nastavenie okrem názvu kanálu distribúcie na to, aby ste začali.</p>
-    <p xml:lang="sl">Konversation je odjemalec za IRC, ki ponuja hiter dostop do kanalov omrežja Freenode, na katerih boste našli pomoč za večino distribucij. Nastavljen je tako, da za začetek ne potrebuje dodatnih nastavitev, ampak le ime kanala vaše distribucije.</p>
+    <p xml:lang="ca">El Konversation és un client d'IRC que dóna accés ràpid als canals de la xarxa Libera, a on trobareu ajuda per a la majoria de les distribucions. De manera predeterminada, per a començar no necessita més configuració que indicar el nom del canal de la distribució.</p>
+    <p xml:lang="ca-valencia">El Konversation és un client d'IRC que dóna accés ràpid als canals de la xarxa Libera, a on trobareu ajuda per a la majoria de les distribucions. De manera predeterminada, per a començar no necessita més configuració que indicar el nom del canal de la distribució.</p>
+    <p xml:lang="da">Konversation er en IRC-klient som giver dig hurtig adgang til kanaler på Libera-netværket, hvor du finder support til de fleste distributioner. Dets standardindstillinger er sådan at der ikke kræves anden konfiguration end at angive navnet på distributionskanalen for at komme i gang.</p>
+    <p xml:lang="de">Konversation ist ein IRC-Programm, mit dem Sie schnell auf Libera-Kanäle zugreifen können, in denen Sie Unterstützung für die meisten Distributionen erhalten. Die Voreinstellungen sind so gewählt, dass Sie nur den Kanalnamen Ihrer Distribution eingeben müssen.</p>
+    <p xml:lang="el">Το Konversation είναι μια εφαρμογή πελάτη IRC, το οποίο παρέχει ταχεία πρόσβαση σε δικτυακά κανάλια Libera, όπου θα βρείτε υποστήριξη για την πλειοψηφία των διανομών. Οι προκαθορισμένες ρυθμίσεις του καθιστούν περιττή την επιπλέον διαμόρφωση, με εξαίρεση το όνομα του καναλιού της διανομής σας για να ξεκινήσει.</p>
+    <p xml:lang="en-GB">Konversation is an IRC client which gives you speedy access to Libera network's channels, where you will find support for most distributions. Its defaults are such that it needs no configuration other than telling it the name of your distribution channel, to get started.</p>
+    <p xml:lang="es">Konversation es un cliente de IRC que le proporciona un acceso rápido a los canales de la red Libera, donde usted encontrará apoyo para muchas distribuciones. Sus opciones por omisión son tales que no necesita otra configuración que decirle el nombre del canal de distribución para empezar.</p>
+    <p xml:lang="et">Konversation on IRC klient, mis lubab sul edukalt tegutseda Libera'i võrgu kanalitel. Rakenduse vaikimisi seadistused on niisugused, et sul pole alustamiseks üldjuhul vaja midagi rohkemat, kui ainult teada kanalit, millega ühineda.</p>
+    <p xml:lang="fi">Konversation on IRC-asiakas, jolla pääset nopeasti Liberan verkkokanaville, mistä löydät useimpien jakelujen tuen. Sen oletusasetukset ovat sellaiset, ettei sinun tarvitse kuin vain kertoa jakelusi kanava päästäksesi alkuun.</p>
+    <p xml:lang="fr">Konversation est un client « IRC » vous donnant un accès rapide aux canaux du réseau Libera où vous trouverez de l'aide pour la plupart des distributions. Ses configurations par défaut sont telles que vous devez uniquement spécifier le nom de votre distribution pour débuter.</p>
+    <p xml:lang="gl">Konversation é un cliente de IRC que lle da acceso rápido ás canles da rede Libera, onde atopará asistencia técnica para a meirande parte das distribucións. A configuración predeterminada está deseñada para que só teña que indicar o nome da canle da distribución para comezar.</p>
+    <p xml:lang="he">שיחה KDE זה לקוח IRC שנותן גישה מהירה לערוצי רשתות Libera, ותומך ברוב ההפצות. כברירת מחדל לא צריך להגדיר חוץ לבחור שם של אחד מערוצי ההפצות בשביל להתחיל.</p>
+    <p xml:lang="id">Konversation adalah klien IRC yang memberimu akses cepat ke channel jaringan Libera, di mana kamu akan menemukan dukungan untuk sebagian besar distribusi. Default-nya adalah tidak diperlukannya konfigurasi selain memberi tahu nama channel distribusimu, untuk memulai.</p>
+    <p xml:lang="it">Konversation è un client IRC che fornisce un acceso rapido ai canali della rete Libera, dove troverai supporto per la maggior parte delle distribuzioni. I suoi valori predefiniti fanno sì che non ci sia bisogno di configurazione ulteriore a parte il nome del canale della distribuzione per poter iniziare.</p>
+    <p xml:lang="ko">Konversation은 여러 IRC 네트워크 채널에 빠르게 접근할 수 있는 IRC 클라이언트입니다. 대부분 리눅스 배포판 사용자와 대화할 수 있는 Libera 네트워크 및 배포판 채널을 기본 설정으로 포함합니다.</p>
+    <p xml:lang="nl">Konversation is een IRC-client die u snel toegang geeft tot networkkanalen van Libera, waar u ondersteuning voor de meeste distributies zult vinden. Zijn standaarden zijn zo dat het geen instellingen nodig heeft anders dan het de naam van uw distributiekanaal te vertellen om van start te gaan.</p>
+    <p xml:lang="nn">Konversation er ein IRC-klient som gjev deg rask tilgang til alle Libera-kanalane, der du finn støtte for dei fleste distribusjonane. Standardverdiane er slik at det ikkje trengst noko vidare oppsett; du treng berre oppgje kanalen til distribusjonen din for å komma i gang.</p>
+    <p xml:lang="pl">Konversation jest klientem IRC, który szybko daje ci dostęp do kanałów sieci Libera, gdzie mozesz znaleźć wsparcie dla większości dystrybucji. Dzięki jego ustawieniom domyślnym wystarczy, że podasz nazwę swojego kanału dystrybucji, aby zacząć go używać.</p>
+    <p xml:lang="pt">O Konversation é um cliente de IRC que lhe dá um acesso rápido aos canais da rede Libera, onde poderá obter suporte para a maioria das distribuições. Tem um conjunto de definições por omissão suficientemente úteis que praticamente não necessita de mais configurações do que definir o nome do canal da sua distribuição para começar.</p>
+    <p xml:lang="pt-BR">O Konversation é um cliente IRC que fornece a você acesso rápido aos canais da rede Libera, onde você irá encontrar suporte para a maioria das distribuições. Seus padrões são tais que ele não precisa de nenhuma configuração além de dizer o nome do canal de sua distribuição para poder começar.</p>
+    <p xml:lang="sk">Konversation je IRC klient, ktorý vám umožní rýchly prístup ku kanálom siete Libera, kde nájdete podporu pre väčšinu distribúcií. Predvolene nepotrebuje žiadne nastavenie okrem názvu kanálu distribúcie na to, aby ste začali.</p>
+    <p xml:lang="sl">Konversation je odjemalec za IRC, ki ponuja hiter dostop do kanalov omrežja Libera, na katerih boste našli pomoč za večino distribucij. Nastavljen je tako, da za začetek ne potrebuje dodatnih nastavitev, ampak le ime kanala vaše distribucije.</p>
     <p xml:lang="sr">Конверзација је ИРЦ клијент који вам омогућава брз приступ каналима Фринодове мреже, где можете наћи подршку за већину дистрибуција. Подразумеване поставке су такве да не треба ништа да подешавате, већ само да унесете име канала дистрибуције, да бисте почели.</p>
-    <p xml:lang="sr-Latn">Konverzacija je IRC klijent koji vam omogućava brz pristup kanalima Freenodeove mreže, gde možete naći podršku za većinu distribucija. Podrazumevane postavke su takve da ne treba ništa da podešavate, već samo da unesete ime kanala distribucije, da biste počeli.</p>
+    <p xml:lang="sr-Latn">Konverzacija je IRC klijent koji vam omogućava brz pristup kanalima Liberaove mreže, gde možete naći podršku za većinu distribucija. Podrazumevane postavke su takve da ne treba ništa da podešavate, već samo da unesete ime kanala distribucije, da biste počeli.</p>
     <p xml:lang="sr-ijekavian">Конверзација је ИРЦ клијент који вам омогућава брз приступ каналима Фринодове мреже, где можете наћи подршку за већину дистрибуција. Подразумеване поставке су такве да не треба ништа да подешавате, већ само да унесете име канала дистрибуције, да бисте почели.</p>
-    <p xml:lang="sr-ijekavianlatin">Konverzacija je IRC klijent koji vam omogućava brz pristup kanalima Freenodeove mreže, gde možete naći podršku za većinu distribucija. Podrazumevane postavke su takve da ne treba ništa da podešavate, već samo da unesete ime kanala distribucije, da biste počeli.</p>
-    <p xml:lang="sv">Konversation är en IRC-klient som ger snabb åtkomst till Freenode-nätverkets kanaler, där det finns stöd för de flesta distributioner. Standardinställningarna är sådana att ingen inställning behövs för att komma igång annat än att tala om namnet på distributionens kanal för den.</p>
-    <p xml:lang="tr">Konversation, çoğu dağıtım için destek bulacağınız Freenode ağ kanallarına hızlı erişim sağlayan bir IRC istemcisidir. Varsayılanları, başlamak için dağıtım kanalınızın adını söylemekten başka bir yapılandırmaya ihtiyaç duymayacak şekildedir.</p>
-    <p xml:lang="uk">Konversation — клієнтська програма IRC, за допомогою якої ви зможете швидко отримати доступ до каналів мережі Freenode, на яких можна спілкуватися із іншими користувачами та розробниками. Якщо ви скористаєтеся типовими параметрами роботи програми, вам достатньо буде вказати лише назву потрібного вам каналу, щоб розпочати спілкування на ньому.</p>
-    <p xml:lang="x-test">xxKonversation is an IRC client which gives you speedy access to Freenode network's channels, where you will find support for most distributions. Its defaults are such that it needs no configuration other than telling it the name of your distribution channel, to get started.xx</p>
-    <p xml:lang="zh-CN">Konversation IRC 客户端让您快速链接 Freenode 网络的频道，以获取大多数发行版的社区支持。默认情况下，您只需要输入发行版频道的名字就可以开始使用了，而不需要额外配置。</p>
-    <p xml:lang="zh-TW">Konversation 是一個讓您可以快速連上 Freenode 網路頻道的 IRC 用戶端，在那裡您可以找到大多數散佈版的支援。其預設不需要任何設定，您只需要告訴它您散佈版頻道的名字就可以開始了。</p>
+    <p xml:lang="sr-ijekavianlatin">Konverzacija je IRC klijent koji vam omogućava brz pristup kanalima Liberaove mreže, gde možete naći podršku za većinu distribucija. Podrazumevane postavke su takve da ne treba ništa da podešavate, već samo da unesete ime kanala distribucije, da biste počeli.</p>
+    <p xml:lang="sv">Konversation är en IRC-klient som ger snabb åtkomst till Libera-nätverkets kanaler, där det finns stöd för de flesta distributioner. Standardinställningarna är sådana att ingen inställning behövs för att komma igång annat än att tala om namnet på distributionens kanal för den.</p>
+    <p xml:lang="tr">Konversation, çoğu dağıtım için destek bulacağınız Libera ağ kanallarına hızlı erişim sağlayan bir IRC istemcisidir. Varsayılanları, başlamak için dağıtım kanalınızın adını söylemekten başka bir yapılandırmaya ihtiyaç duymayacak şekildedir.</p>
+    <p xml:lang="uk">Konversation — клієнтська програма IRC, за допомогою якої ви зможете швидко отримати доступ до каналів мережі Libera, на яких можна спілкуватися із іншими користувачами та розробниками. Якщо ви скористаєтеся типовими параметрами роботи програми, вам достатньо буде вказати лише назву потрібного вам каналу, щоб розпочати спілкування на ньому.</p>
+    <p xml:lang="x-test">xxKonversation is an IRC client which gives you speedy access to Libera network's channels, where you will find support for most distributions. Its defaults are such that it needs no configuration other than telling it the name of your distribution channel, to get started.xx</p>
+    <p xml:lang="zh-CN">Konversation IRC 客户端让您快速链接 Libera 网络的频道，以获取大多数发行版的社区支持。默认情况下，您只需要输入发行版频道的名字就可以开始使用了，而不需要额外配置。</p>
+    <p xml:lang="zh-TW">Konversation 是一個讓您可以快速連上 Libera 網路頻道的 IRC 用戶端，在那裡您可以找到大多數散佈版的支援。其預設不需要任何設定，您只需要告訴它您散佈版頻道的名字就可以開始了。</p>
     <p>Features:</p>
     <p xml:lang="ar">الميزات:</p>
     <p xml:lang="ca">Característiques:</p>
diff --git a/src/config/preferences.cpp b/src/config/preferences.cpp
index 48d9467..8a28caa 100644
--- a/src/config/preferences.cpp
+++ b/src/config/preferences.cpp
@@ -57,9 +57,9 @@ Preferences::Preferences()
     mIdentity->setNicknameList(nickList);
 
     Konversation::ServerGroupSettingsPtr serverGroup(new Konversation::ServerGroupSettings);
-    serverGroup->setName(QStringLiteral("freenode"));
+    serverGroup->setName(QStringLiteral("libera"));
     Konversation::ServerSettings server;
-    server.setHost(QStringLiteral("chat.freenode.net"));
+    server.setHost(QStringLiteral("irc.libera.chat"));
     server.setPort(6697);
     server.setSSLEnabled(true);
     serverGroup->addServer(server);
diff --git a/src/irc/servergroupdialogui.ui b/src/irc/servergroupdialogui.ui
index 4aba886..71dd7c5 100644
--- a/src/irc/servergroupdialogui.ui
+++ b/src/irc/servergroupdialogui.ui
@@ -91,7 +91,7 @@
    <item row="3" column="1" colspan="2">
     <widget class="KLineEdit" name="m_commandEdit">
      <property name="whatsThis">
-      <string>&lt;qt&gt;Optional. This command will be sent to the server after connecting. Example: &lt;b&gt;/msg NickServ IDENTIFY &lt;i&gt;konvirocks&lt;/i&gt;&lt;/b&gt;. This example is for the freenode network, which requires users to register their nickname with a password and login when connecting. &lt;i&gt;konvirocks&lt;/i&gt; is the password for the nickname given in Identity. You may enter more than one command by separating them with semicolons.&lt;/qt&gt;</string>
+      <string>&lt;qt&gt;Optional. This command will be sent to the server after connecting. Example: &lt;b&gt;/msg NickServ IDENTIFY &lt;i&gt;konvirocks&lt;/i&gt;&lt;/b&gt;. This example is for the Libera network, which requires users to register their nickname with a password and login when connecting. &lt;i&gt;konvirocks&lt;/i&gt; is the password for the nickname given in Identity. You may enter more than one command by separating them with semicolons.&lt;/qt&gt;</string>
      </property>
     </widget>
    </item>
diff --git a/src/irc/serverlistdialogui.ui b/src/irc/serverlistdialogui.ui
index 2f80319..14b1eb5 100644
--- a/src/irc/serverlistdialogui.ui
+++ b/src/irc/serverlistdialogui.ui
@@ -83,7 +83,7 @@
       <bool>true</bool>
      </property>
      <property name="whatsThis">
-      <string>This shows the list of configured IRC networks. An IRC network is a collection of cooperating servers. You need only connect to one of the servers in the network to be connected to the entire IRC network. Once connected, Konversation will automatically join the channels shown. When Konversation is started for the first time, the Freenode network and the &lt;i&gt;#kde&lt;/i&gt; channel are already entered for you.</string>
+      <string>This shows the list of configured IRC networks. An IRC network is a collection of cooperating servers. You need only connect to one of the servers in the network to be connected to the entire IRC network. Once connected, Konversation will automatically join the channels shown. When Konversation is started for the first time, the Libera network and the &lt;i&gt;#kde&lt;/i&gt; channel are already entered for you.</string>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAsNeeded</enum>
-- 
2.30.2

